lib/src/merged_tree.rs

```diff
@@ -328,6 +328,29 @@ pub struct TreeDiffEntry {
     pub value: BackendResult<(MergedTreeValue, MergedTreeValue)>,
 }
 
+impl TreeDiffEntry {
+    fn adjust_for_copy_tracking(
+        self,
+        source_tree: &MergedTree,
+        copy_records: &CopyRecords,
+    ) -> TreeDiffEntry {
+        let Some(CopyRecord { source, .. }) = copy_records.for_target(&self.target) else {
+            return self;
+        };
+
+        Self {
+            source: source.clone(),
+            target: self.target,
+            value: match self.value {
+                Ok((_, target_value)) => source_tree
+                    .path_value(source)
+                    .map(|source_value| (source_value, target_value)),
+                Err(e) => Err(e),
+            },
```

Comment:
nit: cam we calculate the new values before we construct `Self` and thereby avoid cloning `source`?

Reply:
no, we have to clone source because it is a borrow from the CopyRecords